src/fidesctl/cli/commands/generate.py

```diff
@@ -75,6 +76,55 @@ def generate_dataset_db(
     )
 
 
+@generate_dataset.group(name="gcp")
+@click.pass_context
+def generate_dataset_gcp(ctx: click.Context) -> None:
+    """
+    Generate fidesctl Dataset resources for Google Cloud Platform
+    """
+
+
+@generate_dataset_gcp.command(name="bigquery")
+@click.pass_context
+@click.argument("dataset_name", type=str)
+@click.argument("output_filename", type=str)
+@credentials_id_option
+@click.option("--keyfile-path", type=str)
+@include_null_flag
+@with_analytics
+def generate_dataset_bigquery(
+    ctx: click.Context,
+    dataset_name: str,
+    output_filename: str,
+    keyfile_path: str,
+    credentials_id: str,
+    include_null: bool,
+) -> None:
+    """
+    Connect to a BigQuery dataset directly via a SQLAlchemy connection and
+    generate a dataset manifest file that consists of every schema/table/field.
+    A path to a google authorization keyfile can be supplied as an option, or a
+    credentials reference to fidesctl config.
+
+    This is a one-time operation that does not track the state of the dataset.
+    It will need to be run again if the dataset schema changes.
+    """
+
+    bigquery_config = handle_bigquery_config_options(
+        fides_config=ctx.obj["CONFIG"],
+        dataset=dataset_name,
+        keyfile_path=keyfile_path,
+        credentials_id=credentials_id,
+    )
+
+    bigquery_engine = get_bigquery_engine(bigquery_config)
+    bigquery_schemas = _dataset.get_db_schemas(engine=bigquery_engine)
```

Comment:
This seems like a bit too much business logic in the CLI command, this should get abstracted into `dataset.py` and/or reuse as much existing code as possible

Reply:
I think that makes sense, I think I leaned more this way due to the need we have had lately to move CLI specific items up instead. Will clean this up